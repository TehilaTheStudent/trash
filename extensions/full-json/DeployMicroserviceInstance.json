{
  "plugin_name": "DeployMicroserviceInstance",
  "display_name": "",
  "data": [
    {
      "unique_id": "4c2f986ccbfe47d2a3e38049f8454628",
      "name": "Environment",
      "default_value": "",
      "plugin_name": "DeployMicroserviceInstance",
      "version": "0.0.10",
      "type": "d-select",
      "workspace_id": "*",
      "layout_content": "[{\"componentName\":\"d-form\",\"id\":1732957293669,\"category\":\"form\",\"props\":{\"layout\":\"vertical\",\"ref\":\"formRef\"},\"children\":[{\"componentName\":\"d-form-item\",\"id\":1732957332605,\"category\":\"form\",\"props\":{\"label\":\"Environment\",\"field\":\"Environment\",\"help-tips\":\"\",\"rules\":[{\"required\":true,\"message\":\"Environment value cannot be left blank.\",\"trigger\":[\"blur\",\"change\"]},{\"regex\":\"\",\"message\":\"Environment Invalid content.\",\"trigger\":[\"blur\",\"change\"]}],\"visible\":{\"type\":\"fixed\",\"conditions\":[],\"value\":true},\"disabled\":false},\"children\":[{\"componentName\":\"d-select\",\"id\":1732957332604,\"props\":{\"selectType\":\"normal\",\"singleOptions\":{\"type\":\"api\",\"fixedOptions\":[{\"label\":\"Option\",\"value\":\"1\",\"defaultChecked\":false,\"visible\":false}],\"apiOptions\":{\"url\":\"/v2/${project_id}/environments\",\"method\":\"GET\",\"params\":{},\"header\":{},\"label\":\"name\",\"value\":\"id\",\"body\":{},\"responseUrl\":\"environments\",\"linkedFields\":[],\"remote\":false,\"remoteName\":\"\",\"remoteQueryField\":\"params\",\"searchTips\":\"\"}},\"isUpdateChildAll\":true,\"remote\":false,\"allow-clear\":false,\"showPlaceholder\":\"\",\"filter\":true,\"disabled\":{\"type\":\"fixed\",\"conditions\":[],\"value\":false}},\"category\":\"form\"}]},{\"componentName\":\"d-form-item\",\"id\":1732957333730,\"category\":\"form\",\"props\":{\"label\":\"Microservice\",\"field\":\"Microservice\",\"help-tips\":\"\",\"rules\":[{\"required\":true,\"message\":\"Microservice value cannot be left blank.\",\"trigger\":[\"blur\",\"change\"]},{\"regex\":\"\",\"message\":\"Microservice Invalid content.\",\"trigger\":[\"blur\",\"change\"]}],\"visible\":{\"type\":\"fixed\",\"conditions\":[],\"value\":true},\"disabled\":false},\"children\":[{\"componentName\":\"d-select\",\"id\":1732957333729,\"props\":{\"selectType\":\"normal\",\"singleOptions\":{\"type\":\"api\",\"fixedOptions\":[{\"label\":\"Option\",\"value\":\"1\",\"defaultChecked\":false,\"visible\":false}],\"apiOptions\":{\"url\":\"/v2/${project_id}/component/list/query\",\"method\":\"POST\",\"params\":{},\"header\":{},\"label\":\"name\",\"value\":\"id\",\"body\":{\"offset\":0,\"limit\":10,\"name\":\"\",\"sort_dir\":\"\"},\"responseUrl\":\"data\",\"linkedFields\":[],\"remote\":false,\"remoteName\":\"\",\"remoteQueryField\":\"params\",\"searchTips\":\"\"}},\"isUpdateChildAll\":true,\"remote\":false,\"allow-clear\":false,\"showPlaceholder\":\"\",\"filter\":true,\"disabled\":{\"type\":\"fixed\",\"conditions\":[],\"value\":false}},\"category\":\"form\"}]},{\"componentName\":\"d-form-item\",\"id\":1748427612274,\"category\":\"form\",\"props\":{\"label\":\"Change Id\",\"field\":\"CHANGE_ID_INPUT_PARAM\",\"help-tips\":\"If left blank, the plugin will automatically use the predefined COMPONENT_CHANGE_IDS parameter. In change-based pipelines it contains a list of change request IDs explicitly associated with the component, In repository-based pipelines, it includes the change request implicitly linked to the repository according to the triggering branch.\",\"visible\":{\"type\":\"fixed\",\"conditions\":[],\"value\":true},\"disabled\":false},\"children\":[{\"componentName\":\"d-input\",\"id\":1748427612273,\"props\":{\"inputType\":\"normal\",\"showPlaceholder\":\"\",\"isTrim\":true,\"autocomplete\":\"off\",\"defaultValue\":\"\",\"disabled\":{\"type\":\"fixed\",\"conditions\":[],\"value\":false}},\"category\":\"form\"}]},{\"componentName\":\"d-form-item\",\"id\":1732957393845,\"category\":\"form\",\"props\":{\"label\":\"Helm Parameters Patch\",\"field\":\"DeploymentManifestPatchParameters\",\"help-tips\":\"Helm Parameters Patch\",\"rules\":[{\"required\":true,\"message\":\"Helm Parameters Patch value cannot be left blank.\",\"trigger\":[\"blur\",\"change\"]},{\"regex\":\"\",\"message\":\"Helm Values Patch Invalid content.\",\"trigger\":[\"blur\",\"change\"]}],\"visible\":{\"type\":\"fixed\",\"conditions\":[],\"value\":true},\"disabled\":false},\"children\":[{\"componentName\":\"key-value\",\"id\":1732957393844,\"props\":{},\"category\":\"form\"}]},{\"componentName\":\"d-form-item\",\"id\":1748414000262,\"category\":\"form\",\"props\":{\"label\":\"Wait Until Deployment Finishes\",\"field\":\"isWait\",\"help-tips\":\"\",\"visible\":{\"type\":\"fixed\",\"conditions\":[],\"value\":true},\"disabled\":false},\"children\":[{\"componentName\":\"d-switch\",\"id\":1748414000261,\"props\":{\"defaultValue\":true,\"disabled\":{\"type\":\"fixed\",\"conditions\":[],\"value\":false}},\"category\":\"form\"}]}]}]",
      "front_data_prop": null,
      "label": "default",
      "description": null,
      "required": null,
      "extend_prop": {
        "api_options": null,
        "api_type": null,
        "show_placeholder": null,
        "options": null,
        "disabled_conditions": null,
        "visible_conditions": null
      },
      "validation": {
        "required_message": null,
        "regex": null,
        "regex_message": null,
        "max_length": null,
        "min_length": null
      }
    },
    {
      "unique_id": "1cfbba95fcd544f9b8f004ece39ee183",
      "name": "Microservice",
      "default_value": "",
      "plugin_name": "DeployMicroserviceInstance",
      "version": "0.0.10",
      "type": "d-select",
      "workspace_id": "*",
      "layout_content": "",
      "front_data_prop": null,
      "label": "default",
      "description": null,
      "required": null,
      "extend_prop": {
        "api_options": null,
        "api_type": null,
        "show_placeholder": null,
        "options": null,
        "disabled_conditions": null,
        "visible_conditions": null
      },
      "validation": {
        "required_message": null,
        "regex": null,
        "regex_message": null,
        "max_length": null,
        "min_length": null
      }
    },
    {
      "unique_id": "f693b82592d348a08726b48d266d5646",
      "name": "CHANGE_ID_INPUT_PARAM",
      "default_value": "",
      "plugin_name": "DeployMicroserviceInstance",
      "version": "0.0.10",
      "type": "d-input",
      "workspace_id": "*",
      "layout_content": "",
      "front_data_prop": null,
      "label": "default",
      "description": null,
      "required": null,
      "extend_prop": {
        "api_options": null,
        "api_type": null,
        "show_placeholder": null,
        "options": null,
        "disabled_conditions": null,
        "visible_conditions": null
      },
      "validation": {
        "required_message": null,
        "regex": null,
        "regex_message": null,
        "max_length": null,
        "min_length": null
      }
    },
    {
      "unique_id": "5014a5cd4ef24cd081f8a615fa37aeb4",
      "name": "DeploymentManifestPatchParameters",
      "default_value": "",
      "plugin_name": "DeployMicroserviceInstance",
      "version": "0.0.10",
      "type": "key-value",
      "workspace_id": "*",
      "layout_content": "",
      "front_data_prop": null,
      "label": "default",
      "description": null,
      "required": null,
      "extend_prop": {
        "api_options": null,
        "api_type": null,
        "show_placeholder": null,
        "options": null,
        "disabled_conditions": null,
        "visible_conditions": null
      },
      "validation": {
        "required_message": null,
        "regex": null,
        "regex_message": null,
        "max_length": null,
        "min_length": null
      }
    },
    {
      "unique_id": "accc7bd3f86a44bca9f07a283563df25",
      "name": "isWait",
      "default_value": "true",
      "plugin_name": "DeployMicroserviceInstance",
      "version": "0.0.10",
      "type": "d-switch",
      "workspace_id": "*",
      "layout_content": "",
      "front_data_prop": null,
      "label": "default",
      "description": null,
      "required": null,
      "extend_prop": {
        "api_options": null,
        "api_type": null,
        "show_placeholder": null,
        "options": null,
        "disabled_conditions": null,
        "visible_conditions": null
      },
      "validation": {
        "required_message": null,
        "regex": null,
        "regex_message": null,
        "max_length": null,
        "min_length": null
      }
    }
  ]
}