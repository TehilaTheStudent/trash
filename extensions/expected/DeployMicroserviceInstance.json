[
  {
    "key": "Environment",
    "value": "f568c0b09c304fe589241d81521d9ba8"
  },
  {
    "key": "Microservice",
    "value": "63a06a7860c7431c8d0cf648578950ea"
  },
  {
    "key": "CHANGE_ID_INPUT_PARAM",
    "value": "${COMPONENT_ID}"
  },
  {
    "key": "DeploymentManifestPatchParameters",
    "value": "[{\"key\":\"image.tag\",\"value\":\"demo\"}]"
  },
  {
    "key": "isWait",
    "value": true
  }
]